firmware/src/powhsm/src/trie.c

```diff
@@ -127,7 +127,7 @@ uint8_t trie_consume(uint8_t *buf, const uint8_t len) {
             ctx->raw_size = 1;
             --ctx->remaining_bytes;
             ctx->state = TRIE_ST_SHARED_PREFIX;
-            if (buf[i] >= 0 && buf[i] <= 31) {
+            if (/*buf[i] >= 0 &&*/ buf[i] <= 31) {
```

Comment:
Again, the `-Wall -Wextra -Werror` compiler flags would cause a compiler error due to `buf` being an `uint8_t` (due to the fact that `buf[i] >= 0` is a tautology). Left it as a comment just as a reminder to the developer. Should I remove it you think?

Reply:
Up to you. If you think having this adds value as a reminder for the future, let's leave it there, it doesn't bother me so much.

I thought it could be some quick test you forgot to clean up. Since it's intentional I'm good with it.